src/cache.cpp

```diff
@@ -803,7 +805,24 @@ void Cache::cleanup_cache(std::vector<std::shared_ptr<RssFeed>> feeds,
 	} else {
 		LOG(Level::DEBUG,
 			"Cache::cleanup_cache: NOT cleaning up cache...");
+
+		std::string query = (
+				"SELECT count(rss) "
+				"FROM ("
+				"SELECT feedurl AS rss FROM rss_item "
+				"UNION ALL "
+				"SELECT rssurl FROM rss_feed"
+				") "
+				"WHERE rss NOT IN "
+			);
+		query.append(list);
+		query.push_back(';');
+
+		CbHandler count_cbh;
+		run_sql(query, count_callback, &count_cbh);
+		unreachable = count_cbh.count();
 	}
+	return unreachable;
```

Comment:
I think the comment about the missing unlock should be moved here, because both `if` branches now contain SQL queries, and the comment applies to both of them.